finance/classes/accounting/Invoice.class.php

```diff
@@ -197,40 +197,34 @@ public static function getColumns() {
         ];
     }
 
-    public static function calcIsPaid($om, $oids, $lang) {
+    public static function calcIsPaid($self) {
         $result = [];
-        // #memo - fundings_ids targets all fundings relating to invoice: this includes the installments
-        // we need to limit the check to the direct funding, if any
-        $invoices = $om->read(get_called_class(), $oids, ['status', 'price', 'funding_id.paid_amount'], $lang);
-        if($invoices > 0) {
-            foreach($invoices as $oid => $invoice) {
-                $result[$oid] = false;
-                if($invoice['status'] != 'invoice') {
-                    // proforma invoices cannot be marked as paid
-                    continue;
-                }
-                if($invoice['price'] == 0) {
-                    // mark the invoice as paid, whatever its funding
-                    $result[$oid] = true;
-                    continue;
-                }
-                if($invoice['funding_id.paid_amount'] && $invoice['funding_id.paid_amount'] == $invoice['price']) {
-                    $result[$oid] = true;
-                }
+        $self -> read(['status', 'price', 'funding_id' =>['paid_amount']]);
+        foreach($self as $id => $invoice) {
+            $result[$id] = false;
+            if($invoice['status'] != 'invoice') {
+                continue;
+            }
+            if($invoice['price'] == 0) {
+                $result[$id] = true;
+                continue;
+            }
+            if($invoice['funding_id'].['paid_amount'] && $invoice['funding_id'].['paid_amount'] == $invoice['price']) {
```

Comment:
A dot left

$invoice['funding_id'].['paid_amount']      should be    -> $invoice['funding_id']['paid_amount']